providers/exoscale/exoscaleProvider.go

```diff
@@ -68,7 +70,8 @@ func (client *exoscaleProvider) GetZoneRecords(domain string) (models.Records, e
 func (c *exoscaleProvider) GetDomainCorrections(dc *models.DomainConfig) ([]*models.Correction, error) {
 	dc.Punycode()
 
-	records, err := c.client.GetRecords(dc.Name)
```

Comment:
You don't have to do this in this PR, but it would be great if you could split out the code that downloads the records and creates RecordConfigs as GetZoneRecords().   That would contribute to https://github.com/StackExchange/dnscontrol/issues/628 (see https://stackexchange.github.io/dnscontrol/get-zones for instructions).

Reply:
Sure, I may add that in a future PR — right now I am focused on trying to get this to be importable into my own project